src/lib/operators/join_nested_loop.hpp

```diff
@@ -22,15 +24,39 @@ class JoinNestedLoop : public AbstractJoinOperator {
       const std::vector<AllParameterVariant>& args, const std::shared_ptr<AbstractOperator>& recreated_input_left,
       const std::shared_ptr<AbstractOperator>& recreated_input_right) const override;
 
+  struct JoinParams {
+    PosList& pos_list_left;
+    PosList& pos_list_right;
+    std::vector<bool>& left_matches;
+    std::vector<bool>& right_matches;
+    bool track_left_matches;
+    bool track_right_matches;
+    JoinMode mode;
+    const PredicateCondition predicate_condition;
+  };
```

Comment:
We could make some (more) of these const as well I guess, like `track_left_matches`, `track_right_matches`, and `mode`, right?